aten/src/ATen/core/ivalue_inl.h

```diff
@@ -227,10 +227,17 @@ struct TORCH_API ConstantString final : c10::intrusive_ptr_target {
 
  public:
   ConstantString(std::string str) : str_(std::move(str)) {}
+  ConstantString(c10::string_view str) : str_(std::string(str)) {}
   static c10::intrusive_ptr<ConstantString> create(std::string str_);
+  static c10::intrusive_ptr<ConstantString> create(c10::string_view str_);
+
   const std::string& string() const {
     return str_;
   }
+  c10::string_view string_view() const {
+    return str_.c_str();
```

Comment:
You should avoid using `c_str -> string_view` as it incurs an unnecessary `strlen` call, I believe you can implicitly convert:
```suggestion
    return str_;
```